sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -81,8 +99,21 @@ case class TungstenProject(projectList: Seq[NamedExpression], child: SparkPlan)
 case class Filter(condition: Expression, child: SparkPlan) extends UnaryNode {
   override def output: Seq[Attribute] = child.output
 
-  protected override def doExecute(): RDD[InternalRow] = child.execute().mapPartitions { iter =>
-    iter.filter(newPredicate(condition, child.output))
+  override lazy val accumulators = Map(
+    "numInputRows" -> sparkContext.internalAccumulator(0L, "number of input rows"),
+    "numOutputRows" -> sparkContext.internalAccumulator(0L, "number of output rows"))
+
+  protected override def doExecute(): RDD[InternalRow] = {
+    val numInputRows = accumulator[Long]("numInputRows")
+    val numOutputRows = accumulator[Long]("numOutputRows")
+    child.execute().mapPartitions { iter =>
+      iter.filter { row =>
+        numInputRows += 1
+        val r = newPredicate(condition, child.output)(row)
```

Comment:
we need to move newPredicate outside of the filter call -- otherwise it is super expensive to create a new predicate every record.